libbfddp/bfddp_extra.h

```diff
@@ -0,0 +1,497 @@
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+/**
+ * \file bfddp_extra.h
+ */
+#ifndef BFDDP_EXTRA_H
+#define BFDDP_EXTRA_H
+
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "bfddp.h"
+#include "bfddp_packet.h"
+
+/** The BFD session structure for holding information. */
+struct bfd_session {
+	/** Peer multiple hop indicator. */
+	bool bs_multihop;
+	/** Demand mode indicator. */
+	bool bs_demand;
+	/** Control Plane Independent indicator. */
+	bool bs_cbit;
+	/** Echo mode indicator. */
+	bool bs_echo;
+	/** Passive mode indicator. */
+	bool bs_passive;
+	/** BFD timers poll indicator. */
+	bool bs_poll;
+	/** BFD final indicator. */
+	bool bs_final;
+	/** BFD administrative shutdown state. */
+	bool bs_admin_shutdown;
+
+	/** IPv4 address indicator. */
+	bool bs_ipv4;
+	/** Local address. */
+	union {
+		struct sockaddr bs_src_sa;
+		struct sockaddr_in bs_src_sin;
+		struct sockaddr_in6 bs_src_sin6;
+	} bs_src;
+	/** Remote address. */
+	union {
+		struct sockaddr bs_dst_sa;
+		struct sockaddr_in bs_dst_sin;
+		struct sockaddr_in6 bs_dst_sin6;
+	} bs_dst;
+
+	/** Local discriminator. */
+	uint32_t bs_lid;
+	/** Remote discriminator. */
+	uint32_t bs_rid;
+
+	/** Desired minimum transmission interval. */
+	uint32_t bs_tx;
+	/** Current desired minimum transmission interval. */
+	uint32_t bs_cur_tx;
+	/** Required minimum receive interval. */
+	uint32_t bs_rx;
+	/** Current required minimum receive interval. */
+	uint32_t bs_cur_rx;
+	/** Required minimum echo receive interval. */
+	uint32_t bs_erx;
+	/** Current required minimum echo receive interval. */
+	uint32_t bs_cur_erx;
+	/** Milliseconds to wait before starting session. */
+	uint32_t bs_hold;
+	/** Detection multiplier. */
+	uint8_t bs_dmultiplier;
+	/** Currently used detection multiplier. */
+	uint8_t bs_cur_dmultiplier;
+
+	/** Minimum amount of TTL to expect. */
+	uint8_t bs_minttl;
+
+	/** Interface index. */
+	uint32_t bs_ifindex;
+	/** Interface name. */
+	char bs_ifname[64];
```

Comment:
might be worth defining `BFD_IF_MAX_SIZE` or s/t like that

Reply:
Makes sense, I'll fix it.